cloudquery/sdk/scalar/list.py

```diff
@@ -33,7 +29,7 @@ def set(self, val: Any):
             self._value = Vector()
             return
 
-        if isinstance(val, Scalar) and type(val) == self._type:
+        if isinstance(val, (Scalar, self._type)):
```

Comment:
`isinstance` works as `one of`.
So, this check will succeed for `Uint` `Scalar` & fail miraculously if the underlying type for `List` is something else.
You should do:
```suggestion
        if isinstance(val, Scalar) and isinstance(val, self._type)):
```

Or, if we're sure that the underlying type for `List` is always `Scalar`, you could do simply:
```suggestion
        if  isinstance(val, self._type)):
```

Reply:
Ahh, I messed up here - I blame the time of day I was working on this section. I misread the previous code's operand as `or` instead of `and`.

The second option is correct in this case, as all valid values for `self._type` here must inherit from `Scalar`.

Thanks for catching it!

_Quick point of clarity for others reading (I realize it's just a typo): `isinstance` uses a short-circuit `any-of`/`or` when supplied with a tuple, not a `one-of`/`xor`._